python/lsst/verify/tasks/metricTask.py

```diff
@@ -244,7 +308,6 @@ def areInputDatasetsScalar(cls, config):
                 for name in connections.inputs}
 
     @classmethod
-    @abc.abstractmethod
     def getOutputMetricName(cls, config):
```

Comment:
I'm worried that that will lead to the kind of problem we have with `CmdLineTask._getConfigName` and `CmdLineTask._getMetadataName`, where it's impossible to determine the associated dataset types without creating an instance of the task (leaving aside the question of whether those methods should have been private in the first place). Even if the _current_ code does not call `getOutputMetricName` externally, I want to retain the ability to do so.

I assume I can't make a property that takes a parameter; that seems pointless syntactically.

Reply:
This static-that-takes-a-config interface is something I picked up from the original `PipelineTask`. Did you run into any problems in initial testing other than the awkwardness of passing in `self.config`?

Reply:
There was one chicken and the egg situation we got into at some point, but I dont remember the details because I didn't run into it personally. Why I do appreciate defensive programming, I also fear creating awkward long lived interfaces because someone one day might need something (see afw). What is the use case you need to see the datasets of a task, without having an instance of the task? Do you imagine tasks overriding this (currently it looks like none do) If they don't, what if it was a free function? Or what if it was a method on the config class?

Reply:
A function was not possible before this PR, but it is now. 🤔